### ISSUE TITLE
TreeList - State Storing

### ISSUE THREAD

**Issue by dxbykov** [DevExpress/DevExtreme#3135]
# The Problem 

While working with the TreeList, end-users usually have it preconfigured (setup filters, column visibility, expand/collapse nodes, etc.) to perform their job tasks. The TreeList is expected to be restored in the same state as it was before navigating to another view or reloading a browser window. Similar functionality is already [implemented in the DataGrid](https://js.devexpress.com/Documentation/17_1/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/) component.

# The Solution 

You can use the `stateStoring` option to enable saving and restoring a TreeList’s state. For instance, you can use the browser’s `localStorage` to save the current TreeList state:

```js
        stateStoring: {
            enabled: true,
            type: 'localStorage',
            storageKey: 'treeListStorage'
        },

```

The TreeList component has the [same API as the DataGrid](https://js.devexpress.com/Documentation/17_1/ApiReference/UI_Widgets/dxDataGrid/Configuration/stateStoring/).

# Try It And Give Us Feedback 

## Live Sandbox

[jQuery](https://codepen.io/dxbykov/pen/ddgVVJ)
[Angular](https://embed.plnkr.co/m8XAxW4gKjUdMBffOZyV/)

## Installation

Follow the installation guide from the [release page](https://github.com/DevExpress/DevExtreme/releases) and leave us feedback on this topic. 

Feel free to leave comments or use [GitHub reactions :thumbsup:](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments).

**Comment by genachka** [comment]
:+1: Thank you for adding this!

**Comment by dxbykov** [comment]
Thanks @genachka and everybody else who gave us early feedback by other means. The feature is now available in the [v18.1](https://github.com/DevExpress/DevExtreme/releases/tag/18.1.3) release. I'm closing this thread. In the case of bugs or questions, feel free to create new GitHub [issues](https://github.com/DevExpress/DevExtreme/issues) or tickets in our [Support Center](https://www.devexpress.com/Support/Center/).
